pkg/apis/operator/v1alpha1/accessor.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+)
+
+// Common conditions.
+const (
+	InstallSucceeded     apis.ConditionType = "InstallSucceeded"
+	DeploymentsAvailable apis.ConditionType = "DeploymentsAvailable"
+)
+
+// Object is a common interface for accessing meta, spec and status of all known types.
+type Object interface {
```

Comment:
As we discussed in slack, I'd like this to be less vague. Here are some suggestions: `Resource`, `CustomResource`, `OperatorInstance`, `OperatorResource`. Maybe `Component` or `KnativeComponent` since I use that word in the docs to describe either eventing or serving. Any of those float your boat?

Reply:
+1 this caught my eye because there's `runtime.Object` upstream.  We also have `duckv1.KResource`.